tests/core/data/test_generic_data_node.py

```diff
@@ -139,28 +168,52 @@ def test_read_write_generic_datanode(self):
         with pytest.raises(MissingWriteFunction):
             generic_dn_3.write(self.data)
 
-    def test_read_write_generic_datanode_with_parameters(self):
+        reset_data()
+
+    def test_read_write_generic_datanode_with_arguments(self):
         generic_dn = GenericDataNode(
             "foo",
             Scope.PIPELINE,
             properties={
-                "read_fct": read_fct_with_params,
-                "write_fct": write_fct_with_params,
-                "read_fct_params": tuple([1]),
-                "write_fct_params": tuple([2]),
+                "read_fct": read_fct_with_args,
+                "write_fct": write_fct_with_args,
+                "read_fct_args": [1],
+                "write_fct_args": [2],
             },
         )
 
         assert all([a + 1 == b for a, b in zip(self.data, generic_dn.read())])
+        assert len(generic_dn.read()) == 10
+
+        generic_dn.write(self.data)
         assert len(generic_dn.read()) == 12
 
+        reset_data()
+
+    def test_read_write_generic_datanode_with_non_list_arguments(self):
+        generic_dn = GenericDataNode(
+            "foo",
+            Scope.PIPELINE,
+            properties={
+                "read_fct": read_fct_with_args,
+                "write_fct": write_fct_with_args,
+                "read_fct_args": 1,
+                "write_fct_args": 2,
```

Comment:
Again, since we check if the read/write args is a List or not in the Config, this should not be able to happen (from the user perspective) right?

Reply:
Yes you're right. The test and the list wrapper as said earlier, is just another layer to avoid system exit error. We alr have checker that will log error about this. This is just a backup of backup :D